sound/soc/sof/ipc.c

```diff
@@ -426,6 +428,9 @@ static void ipc_period_elapsed(struct snd_sof_dev *sdev, u32 msg_id)
 
 	memcpy(&stream->posn, &posn, sizeof(posn));
 
+	/* optionally update position for vBE */
+	dsp_sof_update_guest_posn(sdev, &posn);
+
```

Comment:
conditional compilation.

Reply:
@lgirdwood I'm uncertain about this. I see conditional compilation being used a lot in SOF sources, but actually I tend to agree, that preprocessor conditionals are better in headers, than in .c. In this case it's also easily achievable by simply defining a dummy function in the header for the `#else` case

Reply:
I need a single binary.

Reply:
the only obstacle is NOCODEC, I'm not sure it can be enabled together with real machine drivers now, have to check...